topostats/plotting.py

```diff
@@ -139,31 +139,56 @@ def _outfile(self, plot_suffix: str) -> str:
         """
         return f"{self.stat_to_sum}_{plot_suffix}"
 
-    def sns_plot(self) -> None:
+    def sns_plot(self) -> Optional[Tuple[plt.Figure, plt.Axes]]:
         """Plot the distribution of one or more statistics as either histogram, kernel density estimates or both. Uses
-        base Seaborn."""
+        base Seaborn.
+
+        Returns
+        -------
+        Optional[Union[Tuple[plt.Figure, plt.Axes], None]]
+            Tuple of Matplotlib figure and axes if plotting is successful, None otherwise.
+        """
+
+        # Note: Plotting KDEs with Seaborn is not possible if all values are 0. This is because the KDE is calculated
+        # using a Gaussian kernel and if all values are 0, the standard deviation is 0 wich results in a
+        # ZeroDivisionError. To avoid this, we check if all values are 0 and if so, we skip the KDE plot.
+
         fig, ax = self._setup_figure()
+        LOGGER.info(f"melted data: \n {self.melted_data}")
+        LOGGER.info(f"stat: {self.stat}")
+        LOGGER.info(f"label: {self.label}")
+
+        # If histogram is requested but KDE is not, plot histogram
         if self.hist and not self.kde:
             outfile = self._outfile("hist")
             sns.histplot(data=self.melted_data, x="value", bins=self.bins, stat=self.stat, hue=self.hue)
-        if self.kde and not self.hist:
-            outfile = self._outfile("kde")
-            sns.kdeplot(data=self.melted_data, x="value", hue=self.hue)
-        if self.hist and self.kde:
-            outfile = self._outfile("hist_kde")
-            sns.histplot(
-                data=self.melted_data,
-                x="value",
-                bins=self.bins,
-                stat=self.stat,
-                hue=self.hue,
-                kde=True,
-                kde_kws={"cut": self.cut},
+        # Otherwise, KDE is requested, so check if all values are 0
+        elif (self.melted_data["value"] == 0).all():
```

Comment:
This is an awkward situation but you would also get errors raised if all values were `!= 0` but all the same, because the variance (and in turn the standard deviation which is the square-root of the variance) would be zero.

So a better test if sticking with this structure rather than the suggested `try: ... except:` would be...

```suggestion
        elif self.melted_data["value"].std() == 0:
```